src/core/runtime/projection.cc

```diff
@@ -123,6 +123,23 @@ AffineFunctor<SRC_DIM, TGT_DIM>::AffineFunctor(Runtime* runtime,
   : LegateProjectionFunctor(runtime), transform_(create_transform(dims, weights))
 {
   for (int32_t dim = 0; dim < TGT_DIM; ++dim) offsets_[dim] = offsets[dim];
+
+  // mapping to a different dimension
+  if (SRC_DIM != TGT_DIM) set_collective();
```

Comment:
I assume `set_collective` is monotonic, so you can just return after calling it.